src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -159,8 +159,12 @@ namespace System.Diagnostics
     ///
     /// See the DiagnosticSourceEventSourceBridgeTest.cs for more explicit examples of using this bridge.
     /// </summary>
+#pragma warning disable ESGEN001 // EventSource classes should not declare constructors. 
+                                 // This EventSource uses EtwSelfDescribingEventFormat which is not supported by the source generator yet.
```

Comment:
These comments are not capturing the reasons why these EventSources use `EtwSelfDescribingEventFormat` and not just using the default.

My earlier suggestion was about capturing the reasons.

Reply:
@jkotas I had to move the comment up since I changed the location of the warning to be on the class level that's why I didn't apply those suggestions. Just changed to mention the IEnumerable, thanks!

The reason I changed the location of the warning because I also had to warn on missing `partial` to avoid touching `ref` and `tests`. And, well, the previous warning didn't light up on all ctors if an ES had more than one.

Reply:
ok, moved to ctors